lib/api/readable.js

```diff
@@ -160,6 +163,10 @@ module.exports = class BodyReadable extends Readable {
     }
 
     return await new Promise((resolve, reject) => {
+      if (this[kContentLength] > limit) {
+        this.destroy(new AbortError())
```

Comment:
Can we give theAbortError a message like content length exceeded?

Reply:
actually... dont you need to reject?

something like


  ```js
  return await new Promise((resolve, reject) => {
      if (this[kContentLength]) {
        const contentLength = +this[kContentLength]
        if (
          Number.isFinite(contentLength) &&
          contentLength > limit
        ) {
          const abortError = new AbortError('content length exceeded limit')
          this.destroy(abortError)
          reject(abortError)
          return
        }
      }

```

Reply:
Not strictly necessary.